src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,54 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+/**
+ * driver adapter to drawer with several bugs.
+ */
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
+    private final DrawPanelController dpc;
+    private int startX = 0, startY = 0;
+    private Modes mode = Modes.BASIC;
+
+    public LineDrawerAdapter(DrawPanelController dpc) {
+        super();
+        this.dpc = dpc;
+    }
+
+
+    public void setMode(Modes mode) {
+        this.mode = mode;
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line;
+
+        if (mode == Modes.DOTTED) {
+            line = LineFactory.getDottedLine();
+        } else if (mode == Modes.SPECIAL) {
+            line = LineFactory.getSpecialLine();
+        } else {
+            line = LineFactory.getBasicLine();
+        }
```

Comment:
Unfortunately, this is not a generic solution. (e.g. new line type requires modifying this class, unnecessary direct coupling with LineFactory)